ngi_pipeline/tests/engines/piper_ngi/test_command_creation_config.py

```diff
@@ -53,16 +71,35 @@ def test_build_setup_xml(self):
 
 
     def test_build_piper_cl(self):
-        cl = build_piper_cl(project=self.project_obj, workflow_name=self.workflow_name,
-                            setup_xml_path=self.xml_path, exit_code_path=self.exit_file,
-                            config=self.config, exec_mode='sbatch')
 
-        tcl = cl.split(" ")
+        def _validate_cl(tcl):
+            assert tcl[0] == 'piper'
+            assert '--xml_input' in tcl
+            assert tcl[tcl.index('--xml_input')+1] == self.xml_path
+            assert '--output_directory' in tcl
+            assert tcl[tcl.index('--output_directory')+1] == \
+                    os.path.join('$SNIC_TMP', 'ANALYSIS', self.proj_name, 'piper_ngi')
+            assert '--variant_calling' in tcl
+
+        kwargs = {'project': self.project_obj,
+                  'workflow_name': self.workflow_name,
+                  'setup_xml_path': self.xml_path,
+                  'exit_code_path': self.exit_file,
+                  'config': self.config,
+                  'exec_mode': 'sbatch'}
+
+        cl = build_piper_cl(**kwargs).split(" ")
+        _validate_cl(cl)
+        assert '--keep_pre_bqsr_bam' in cl
+
+        cl = build_piper_cl(generate_bqsr_bam=True, **kwargs).split(" ")
+        _validate_cl(cl)
+        assert '--keep_pre_bqsr_bam' not in cl
+
 
-        assert tcl[0] == 'piper'
-        assert '--xml_input' in tcl
-        assert tcl[tcl.index('--xml_input')+1] == self.xml_path
-        assert '--output_directory' in tcl
-        assert tcl[tcl.index('--output_directory')+1] == \
-                os.path.join('$SNIC_TMP', 'ANALYSIS', self.proj_name, 'piper_ngi')
-        assert '--variant_calling' in tcl
+    def test_analyze(self):
+        with mock.patch('ngi_pipeline.engines.piper_ngi.utils.CharonSession',
+                        spec=False,
+                        return_value=self.charon_mock) as dbmock:
+            analyze(self.project_obj, self.sample_obj)
+        return True
```

Comment:
thanks, just testing a tiny piece of the code though, I'd really wish we'd have proper unit tests.